compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1001,6 +1002,14 @@ impl<T> Binder<T> {
         Binder(f(self.0))
     }
 
+    /// Wraps a `value` in a binder, using the same bound variables as the
+    /// current `Binder`. This should not be used if the new value *changes*
+    /// the bound variables. Note: the (old or new) value itself does not
+    /// necessarily need to *name* all the bound variables.
+    pub fn rebind<U>(&self, value: U) -> Binder<U> {
```

Comment:
It might make sense to add a comment here which says that we do not yet track bound variables.

I am also kind of suprised that none of the methods on `ty::Binder` are marked as inline even though they are very frequently used and really simple. Might be interesting to check if that has a perf impact.

Reply:
I will add a comment. As I said in other comment, leaving inline to another PR.